src/core/ext/transport/chttp2/transport/internal.h

```diff
@@ -501,6 +504,11 @@ struct grpc_chttp2_stream {
   bool write_closed;
   /** Is this stream reading half-closed. */
   bool read_closed;
+  /** Is this stream considered fully closed. This is to avoid race between
```

Comment:
How is this different to `(read_closed && write_closed)`?

Reply:
This is to differentiate between when you consider the stream closed, i.e., just before sending the close message, and when you are done writing to the stream after which you can remove the stream from the map.